src/ateles.erl

```diff
@@ -85,6 +88,35 @@ map_docs({CtxId, _} = Ctx, Docs) ->
     end, Docs)}.
 
 
+acquire_context() ->
+    CtxId = fabric2_util:uuid(),
+    InitClosure = fun(_Ctx) -> ok end,
+    ateles_server:acquire(CtxId, InitClosure).
+
+
+release_context(Ctx) ->
+    ateles_server:release(Ctx).
+
+try_compile(Ctx, FunType, FunName, RawFunSrc) ->
+    FunSrc = case ateles_util:rewrite(Ctx, RawFunSrc) of
+        {ok, RewriteSrc} -> 
+            RewriteSrc;
+        {error, {_Status, RewriteResult}} ->
```

Comment:
How do we handle a case when `ibrowse:send_req_direct` called from https://github.com/cloudant-labs/ateles/blob/master/src/ateles_util.erl#L124 returns `{error, Reason}` see https://github.com/cloudant-labs/ateles/blob/master/src/ateles_util.erl#L135